### PR TITLE
FMD-226 Add TF ProgrammeManagement DB table (Funding section B2)

### PR DESCRIPTION
**Pull request by nsnape**
https://dluhcdigital.atlassian.net/browse/FMD-226

### Change description
A new DB table definition + migration to store Section B2 from TF funding tab(specifically for Towns Deal returns)
Updated ingest pipelines/schemas/mappings, DB, serialisation, and tests.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Run locally, ingest some TD and HS fund files. Extract should now show this data, but only for TD forms.

### CODE REVIEW

File: core/controllers/mappings.py [hunk 1]
```diff
@@ -533,4 +533,30 @@ def get_row_id(model: Type[BaseModel], lookups: dict[str, Any]) -> str | None:
...
+        table="Programme Management",
+        model=ents.ProgrammeManagement,
+        column_mapping={
+            "Submission ID": "submission_id",
+            "Programme ID": "programme_id",
+            "Payment Type": "payment_type",
+            "Spend for Reporting Period": "spend_for_reporting_period",
+            "Start_Date": "start_date",
+            "End_Date": "end_date",
+            "Actual/Forecast": "forecast_or_actual",
```
Comment (Joel-Pearce) [inline core/controllers/mappings.py:546]: Can we standardise this to ```state```, as it appears in Outcomes and Outputs?

**Review by Joel-Pearce — CHANGES_REQUESTED** [review]
"Programme Management" needs to be extracted out of TF R4 as well, right? I may be mistaken, but I think this only extracts the table for Round 3. Could we have some testing on both rounds to ensure that the table is now present when a sheet is ingested for either?

File: core/controllers/mappings.py (hunk 1 above)
Comment (nsnape) [reply core/controllers/mappings.py:546]: yeah I was going to ask about that, this was my 50/50 guess 😆

File: core/serialisation/data_serialiser.py [hunk 2]
```diff
@@ -306,6 +309,25 @@ class Meta:
...
+class ProgrammeManagementSchema(SQLAlchemySchema):
+    """Serialise ProgrammeManagement data"""
+
+    class Meta:
+        model = ProgrammeManagement
+
+    submission_id = auto_field(model=Submission, data_key="SubmissionID")
+    programme_id = auto_field(model=Programme, data_key="ProgrammeID")
+    payment_type = fields.String(attribute="data_blob.payment_type", data_key="PaymentType")
+    spend_for_reporting_period = fields.Integer(
```
Comment (Joel-Pearce) [inline core/serialisation/data_serialiser.py:321]: For other numerical fields we're serialising out of the data blob we use ```fields.Number```. I'm not sure if there's a meaningful difference, but we should probably standardise or specify why different?

**Review by Joel-Pearce — CHANGES_REQUESTED** [review]

File: core/serialisation/data_serialiser.py (hunk 2 above)
Comment (nsnape) [reply core/serialisation/data_serialiser.py:321]: good spot. I'll check the difference in the docs and choose one method for everything

Comment (nsnape) [reply core/serialisation/data_serialiser.py]: Changed this one to Float. Will fix the other existing ones in a separate PR

File: core/serialisation/data_serialiser.py [hunk 3]
```diff
@@ -306,6 +309,25 @@ class Meta:
...
+class ProgrammeManagementSchema(SQLAlchemySchema):
+    """Serialise ProgrammeManagement data"""
+
+    class Meta:
+        model = ProgrammeManagement
+
+    submission_id = auto_field(model=Submission, data_key="SubmissionID")
+    programme_id = auto_field(model=Programme, data_key="ProgrammeID")
+    payment_type = fields.String(attribute="data_blob.payment_type", data_key="PaymentType")
+    spend_for_reporting_period = fields.Float(
```
Comment (Joel-Pearce) [inline core/serialisation/data_serialiser.py:321]: Can we change all of the numbers to ```fields.Float``` when serialised out of the blob now? Or else we have some as ```Number``` and it's not clear why

**Review by Joel-Pearce — CHANGES_REQUESTED** [review]

File: core/serialisation/data_serialiser.py (hunk 3 above)
Comment (nsnape) [reply core/serialisation/data_serialiser.py:321]: yeah, as above comment I'm fixing in a separate PR

**Review by Joel-Pearce — APPROVED** [review]
LGTM

**Pull request merged**